diff --git a/inst/include/simde/arm/neon/qrshl.h b/inst/include/simde/arm/neon/qrshl.h
index 74b8f47..0a68ef9 100644
--- a/inst/include/simde/arm/neon/qrshl.h
+++ b/inst/include/simde/arm/neon/qrshl.h
@@ -192,7 +192,6 @@ simde_vqrshlb_u8(uint8_t a, int8_t b) {
     #elif HEDLEY_HAS_WARNING("-Wsign-conversion")
       /* https://github.com/llvm/llvm-project/commit/f0a78bdfdc6d56b25e0081884580b3960a3c2429 */
       HEDLEY_DIAGNOSTIC_PUSH
-      #pragma clang diagnostic ignored "-Wsign-conversion"
       return vqrshlb_u8(a, b);
       HEDLEY_DIAGNOSTIC_POP
     #else
@@ -234,7 +233,6 @@ simde_vqrshlh_u16(uint16_t a, int16_t b) {
       return vqrshlh_u16(a, HEDLEY_STATIC_CAST(uint16_t, b));
     #elif HEDLEY_HAS_WARNING("-Wsign-conversion")
       HEDLEY_DIAGNOSTIC_PUSH
-      #pragma clang diagnostic ignored "-Wsign-conversion"
       return vqrshlh_u16(a, b);
       HEDLEY_DIAGNOSTIC_POP
     #else
@@ -277,7 +275,6 @@ simde_vqrshls_u32(uint32_t a, int32_t b) {
       return vqrshls_u32(a, HEDLEY_STATIC_CAST(uint16_t, b));
     #elif HEDLEY_HAS_WARNING("-Wsign-conversion")
       HEDLEY_DIAGNOSTIC_PUSH
-      #pragma clang diagnostic ignored "-Wsign-conversion"
       return vqrshls_u32(a, b);
       HEDLEY_DIAGNOSTIC_POP
     #else
@@ -323,7 +320,6 @@ simde_vqrshld_u64(uint64_t a, int64_t b) {
       return vqrshld_u64(a, HEDLEY_STATIC_CAST(uint16_t, b));
     #elif HEDLEY_HAS_WARNING("-Wsign-conversion")
       HEDLEY_DIAGNOSTIC_PUSH
-      #pragma clang diagnostic ignored "-Wsign-conversion"
       return vqrshld_u64(a, b);
       HEDLEY_DIAGNOSTIC_POP
     #else
diff --git a/inst/include/simde/arm/neon/qshl.h b/inst/include/simde/arm/neon/qshl.h
index a01556e..14c9ea4 100644
--- a/inst/include/simde/arm/neon/qshl.h
+++ b/inst/include/simde/arm/neon/qshl.h
@@ -179,7 +179,6 @@ simde_vqshlb_u8(uint8_t a, int8_t b) {
     #elif HEDLEY_HAS_WARNING("-Wsign-conversion")
       /* https://github.com/llvm/llvm-project/commit/f0a78bdfdc6d56b25e0081884580b3960a3c2429 */
       HEDLEY_DIAGNOSTIC_PUSH
-      #pragma clang diagnostic ignored "-Wsign-conversion"
       return vqshlb_u8(a, b);
       HEDLEY_DIAGNOSTIC_POP
     #else
@@ -222,7 +221,6 @@ simde_vqshlh_u16(uint16_t a, int16_t b) {
       return vqshlh_u16(a, HEDLEY_STATIC_CAST(uint16_t, b));
     #elif HEDLEY_HAS_WARNING("-Wsign-conversion")
       HEDLEY_DIAGNOSTIC_PUSH
-      #pragma clang diagnostic ignored "-Wsign-conversion"
       return vqshlh_u16(a, b);
       HEDLEY_DIAGNOSTIC_POP
     #else
@@ -265,7 +263,6 @@ simde_vqshls_u32(uint32_t a, int32_t b) {
       return vqshls_u32(a, HEDLEY_STATIC_CAST(uint16_t, b));
     #elif HEDLEY_HAS_WARNING("-Wsign-conversion")
       HEDLEY_DIAGNOSTIC_PUSH
-      #pragma clang diagnostic ignored "-Wsign-conversion"
       return vqshls_u32(a, b);
       HEDLEY_DIAGNOSTIC_POP
     #else
@@ -308,7 +305,6 @@ simde_vqshld_u64(uint64_t a, int64_t b) {
       return vqshld_u64(a, HEDLEY_STATIC_CAST(uint16_t, b));
     #elif HEDLEY_HAS_WARNING("-Wsign-conversion")
       HEDLEY_DIAGNOSTIC_PUSH
-      #pragma clang diagnostic ignored "-Wsign-conversion"
       return vqshld_u64(a, b);
       HEDLEY_DIAGNOSTIC_POP
     #else
diff --git a/inst/include/simde/wasm/simd128.h b/inst/include/simde/wasm/simd128.h
index bfbf6d3..425ffa3 100644
--- a/inst/include/simde/wasm/simd128.h
+++ b/inst/include/simde/wasm/simd128.h
@@ -8496,7 +8496,6 @@ simde_wasm_f32x4_convert_i32x4 (simde_v128_t a) {
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       HEDLEY_DIAGNOSTIC_PUSH
       #if HEDLEY_HAS_WARNING("-Wc11-extensions")
-        #pragma clang diagnostic ignored "-Wc11-extensions"
       #endif
       r_.altivec_f32 = vec_ctf(a_.altivec_i32, 0);
       HEDLEY_DIAGNOSTIC_POP
diff --git a/inst/include/simde/x86/mmx.h b/inst/include/simde/x86/mmx.h
index fe4914a..13061cf 100644
--- a/inst/include/simde/x86/mmx.h
+++ b/inst/include/simde/x86/mmx.h
@@ -667,7 +667,6 @@ simde_mm_cvtm64_si64 (simde__m64 a) {
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       HEDLEY_DIAGNOSTIC_PUSH
       #if HEDLEY_HAS_WARNING("-Wvector-conversion") && SIMDE_DETECT_CLANG_VERSION_NOT(10,0,0)
-        #pragma clang diagnostic ignored "-Wvector-conversion"
       #endif
       return vget_lane_s64(a_.neon_i64, 0);
       HEDLEY_DIAGNOSTIC_POP
@@ -741,7 +740,6 @@ simde_mm_cvtsi64_si32 (simde__m64 a) {
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       HEDLEY_DIAGNOSTIC_PUSH
       #if HEDLEY_HAS_WARNING("-Wvector-conversion") && SIMDE_DETECT_CLANG_VERSION_NOT(10,0,0)
-        #pragma clang diagnostic ignored "-Wvector-conversion"
       #endif
       return vget_lane_s32(a_.neon_i32, 0);
       HEDLEY_DIAGNOSTIC_POP
@@ -1391,7 +1389,6 @@ simde_mm_sll_pi16 (simde__m64 a, simde__m64 count) {
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       HEDLEY_DIAGNOSTIC_PUSH
       #if HEDLEY_HAS_WARNING("-Wvector-conversion") && SIMDE_DETECT_CLANG_VERSION_NOT(10,0,0)
-        #pragma clang diagnostic ignored "-Wvector-conversion"
       #endif
       r_.neon_i16 = vshl_s16(a_.neon_i16, vmov_n_s16(HEDLEY_STATIC_CAST(int16_t, cnt)));
       HEDLEY_DIAGNOSTIC_POP
@@ -1431,7 +1428,6 @@ simde_mm_sll_pi32 (simde__m64 a, simde__m64 count) {
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       HEDLEY_DIAGNOSTIC_PUSH
       #if HEDLEY_HAS_WARNING("-Wvector-conversion") && SIMDE_DETECT_CLANG_VERSION_NOT(10,0,0)
-        #pragma clang diagnostic ignored "-Wvector-conversion"
       #endif
       r_.neon_i32 = vshl_s32(a_.neon_i32, vmov_n_s32(HEDLEY_STATIC_CAST(int32_t, cnt)));
       HEDLEY_DIAGNOSTIC_POP
@@ -2368,7 +2364,6 @@ simde_m_to_int (simde__m64 a) {
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       HEDLEY_DIAGNOSTIC_PUSH
       #if HEDLEY_HAS_WARNING("-Wvector-conversion") && SIMDE_DETECT_CLANG_VERSION_NOT(10,0,0)
-        #pragma clang diagnostic ignored "-Wvector-conversion"
       #endif
       return vget_lane_s32(a_.neon_i32, 0);
       HEDLEY_DIAGNOSTIC_POP
diff --git a/inst/include/simde/x86/sse2.h b/inst/include/simde/x86/sse2.h
index 395d391..a524960 100644
--- a/inst/include/simde/x86/sse2.h
+++ b/inst/include/simde/x86/sse2.h
@@ -3153,7 +3153,6 @@ simde_mm_cvtepi32_ps (simde__m128i a) {
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       HEDLEY_DIAGNOSTIC_PUSH
       #if HEDLEY_HAS_WARNING("-Wc11-extensions")
-        #pragma clang diagnostic ignored "-Wc11-extensions"
       #endif
       r_.altivec_f32 = vec_ctf(a_.altivec_i32, 0);
       HEDLEY_DIAGNOSTIC_POP
